spec/Functions.tex

```diff
@@ -717,24 +717,29 @@ \subsection{The Ref Return Intent}
 
 When using a \chpl{ref} return intent, the function call is an lvalue
 (specifically, a call expression for a procedure and an iterator variable for
-an iterator).  This implies that the function produces a reference; however,
-this reference cannot be captured.
-%TODO: is this still true with ref variables?
+an iterator).
 
 The \chpl{ref} return intent is specified by following the argument list with
 the \chpl{ref} keyword.  The function must return or yield an lvalue.
-% Todo: disallow void return types for variable functions?
 
-When a procedure with a \chpl{ref} return intent is called on the left-hand
-side of an assignment statement or in the context of a call to a formal
-argument by \chpl{out}, \chpl{inout}, or \chpl{ref} intent, the lvalue that is
-returned by the procedure is assigned a value.
+During function resolution, if a procedure with a \chpl{ref} return
+intent and a procedure without a \chpl{ref} return intent are both
+possible candidates, and either neither use promotion or both do,
```

Comment:
How about: "either both use promotion or neither do"

Also next line: fo -> of